libflux/src/flux/semantic/builtins.rs

```diff
@@ -0,0 +1,338 @@
+                 "truncate" => Node::Builtin("forall [] (t: time, unit: duration) -> time"),
+            }),
+            "experimental" => Node::Package(maplit::hashmap! {
+                 "bigtable" => Node::Package(maplit::hashmap! {
+                     "from" => Node::Builtin("forall [t0] (token: string, project: string, instance: string, table: string) -> [t0]"),
+                 }),
+                 "http" => Node::Package(maplit::hashmap! {
+                     "get" => Node::Builtin("forall [t0, t1] (url: string, ?headers: t0, ?timeout: duration) -> {statusCode: int | body: bytes | headers: t1}"),
+                 }),
+                 "mqtt" => Node::Package(maplit::hashmap! {
+                     "to" => Node::Builtin(r#"
+                         forall [t0, t1] (
+                             <-tables: [t0],
+                             broker: string,
+                             ?topic: string,
+                             ?message: string,
+                             ?qos: int,
+                             ?clientid: string,
+                             ?username: string,
+                             ?password: string,
+                             ?name: string,
+                             ?timeout: duration,
+                             ?timeColumn: string,
+                             ?tagColumns: [string],
+                             ?valueColumns: [string]
+                         ) -> [t1]
+                     "#),
+                 }),
+                 "prometheus" => Node::Package(maplit::hashmap! {
+                     "scrape" => Node::Builtin("forall [t0] (url: string) -> [t0]"),
+                 }),
+                 "addDuration" => Node::Builtin("forall [] (d: duration, to: time) -> time"),
+                 "subDuration" => Node::Builtin("forall [] (d: duration, from: time) -> time"),
+                 "group" => Node::Builtin("forall [t0, t1] (<-tables: [t0], mode: string, columns: [string]) -> t1"),
+                 "objectKeys" => Node::Builtin("forall [t0] (o: t0) -> [string]"),
+                 "set" => Node::Builtin("forall [t0, t1, t2] (<-tables: t0, o: t1) -> t2"),
+                 // must specify exactly one of bucket, bucketID
+                 // must specify exactly one of org, orgID
+                 // if host is specified, token must be too.
+                 "to" => Node::Builtin("forall [t0] (?bucket: string, ?bucketID: string, ?org: string, ?orgID: string, ?host: string, ?token: string) -> t0"),
+            }),
+            "generate" => Node::Package(maplit::hashmap! {
+                "from" => Node::Builtin("forall [] (start: time, stop: time, count: int, fn: (n: int) -> int) -> [{ _start: time | _stop: time | _time: time | _value:int }]"),
+            }),
+            "http" => Node::Package(maplit::hashmap! {
+                "post" => Node::Builtin("forall [t0] (url: string, ?headers: t0, ?data: bytes) -> int"),
```

Comment:
It seems like the `headers` field should be an object. If so, we'll need to create an issue for expressing that constraint.

Reply:
I filed this issue: https://github.com/influxdata/flux/issues/2231